kolibri/core/assets/src/views/k-grid/item.vue

```diff
@@ -1,34 +1,215 @@
 <template>
 
-  <div :class="unitClass">
-    <slot></slot>
+  <div :class="unitClass" :style="computedStyle">
+    <div :class="{ debug: gridMetrics.debug }">
+      <slot></slot>
+    </div>
   </div>
 
 </template>
 
 
 <script>
 
+  import responsiveWindow from 'kolibri.coreVue.mixins.responsiveWindow';
+  import logger from 'kolibri.lib.logging';
+
+  const logging = logger.getLogger(__filename);
+
+  function checkNumber(value) {
+    if (isNaN(value)) {
+      logging.error(`Size (${value}) is not a number`);
+      return false;
+    }
+    const size = parseInt(value);
+    if (size !== Number(value)) {
+      logging.error(`Size (${value}) is not an integer`);
+      return false;
+    }
+    if (size < 1) {
+      logging.error(`Size (${value}) is invalid`);
+      return false;
+    }
+    return true;
+  }
+
+  function checkAlignment(value) {
+    if (!['right', 'center', 'left'].includes(value)) {
+      logging.error(`Alignment must be one of left, right, or center`);
+      return false;
+    }
+    return true;
+  }
+
+  function checkArray(value, validator) {
+    if (value.length !== 3) {
+      logging.error(`Array must have 3 values for small, medium, and large screens`);
+      return false;
+    }
+    for (let i = 0; i < 3; i++) {
+      if (!validator(value[i])) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  function checkInputs(singularValue, arrayValue, propName) {
+    if (singularValue !== undefined && arrayValue !== undefined) {
+      logging.error(`Pass either a single item or an array, but not both for '${propName}'`);
+    }
+  }
+
+  function parseArray(value) {
+    if (value === undefined || Array.isArray(value)) {
+      return value;
+    }
+    // assume it's a string based on input validation
+    return value.split(',').map(val => val.trim());
+  }
+
+  /**
+   * Grid layout items
+   */
   export default {
     name: 'kGridItem',
+    mixins: [responsiveWindow],
     props: {
+      /**
+       * Integer size of the grid item.
+       * Represents either number of columns or a percentage.
+       */
       size: {
         type: [Number, String],
-        required: true,
+        required: false,
+        validator: checkNumber,
       },
-      cols: {
-        type: [Number, String],
-        required: true,
+      /**
+       * Array of integer sizes of the grid item for small, medium, and large screens.
+       * Represents either number of columns or a percentage.
+       */
+      sizes: {
+        type: [Array, String],
+        required: false,
+        validator(value) {
+          return checkArray(parseArray(value), checkNumber);
+        },
+      },
+      /**
+       * When true, sizes are interpretted as a percentage.
+       * Only 25, 50, 75, and 100 are allowed.
```

Comment:
Why limit the percentages to these 4 values?

Reply:
There's a lot more to explain in more detail later, but at a high level:

* The responsive grid has 4, 8, or 12 columns depending on whether the window is small, medium, or large.
* The percentages are intended to work with the responsive grid, regardless of the current breakpoint

For more granular control you can specify an exact number of columns for a particular breakpoint ranges - for example `sizes="3, 7, 11"` would make the item 3/4, 7/8, and 11/12 columns at small, medium, and large screen sizes, respectively.

In practice what I've seen to be more common is that at small sizes an item might be full width, and at larger sizes it might become half or quarter-width For example:

```html
<k-grid-item sizes="100, 50, 50" percentage> ... </k-grid-item>
```